bindings/python3/econf.py

```diff
@@ -0,0 +1,996 @@
+        _exceptions(err, f"get_string_value_def failed with error: {err_string(err)}")
+    return c_result.value.decode("utf-8")
+
+
+def get_bool_value_def(ef: EconfFile, group: str, key: str, default: bool) -> bool:
+    """
+    Return a boolean value for given group/key or return a default value if key is not found
+
+    :param ef: Key-Value storage object
+    :param group: desired group
+    :param key: key of the value that is requested
+    :param default: value to be returned if no key is found
+    :return: value of the key
+    """
+    if group:
+        group = _encode_str(group)
+    c_key = _encode_str(key)
+    c_result = c_bool()
+    if not isinstance(default, bool):
+        raise TypeError('"value" parameter must be of type bool')
+    c_default = c_bool(default)
+    err = LIBECONF.econf_getBoolValueDef(
+        ef._ptr, group, c_key, byref(c_result), c_default
+    )
+    if err:
+        if Econf_err(err) == Econf_err.ECONF_NOKEY:
+            return c_default.value
+        _exceptions(err, f"get_bool_value_def failed with error: {err_string(err)}")
+    return c_result.value
+
+
+def set_int_value(ef: EconfFile, group: str, key: str, value: int) -> None:
+    """
+    Setting an integer value for given group/key
+
+    :param ef: Key-Value storage object
+    :param group: desired group
+    :param key: key of the value that is requested
+    :param value: value to be set for given key
+    :return: Nothing
+    """
+    if group:
+        group = _encode_str(group)
+    c_key = _encode_str(key)
+    c_value = _ensure_valid_int(value)
+    err = LIBECONF.econf_setInt64Value(ef._ptr, group, c_key, c_value)
+    if err:
+        _exceptions(err, f"set_int64_value failed with error: {err_string(err)}")
+
+
+def set_uint_value(ef: EconfFile, group: str, key: str, value: int) -> None:
+    """
+    Setting an unsigned integer value for given group/key
+
+    :param ef: Key-Value storage object
+    :param group: desired group
+    :param key: key of the value that is requested
+    :param value: value to be set for given key
+    :return: Nothing
+    """
+    if group:
+        group = _encode_str(group)
+    c_key = _encode_str(key)
+    c_value = _ensure_valid_uint(value)
+    err = LIBECONF.econf_setUInt64Value(ef._ptr, group, c_key, c_value)
+    if err:
+        _exceptions(err, f"set_uint64_value failed with error: {err_string(err)}")
+
+
+def set_float_value(ef: EconfFile, group: str, key: str, value: float) -> None:
+    """
+    Setting a float value for given group/key
+
+    :param ef: Key-Value storage object
+    :param group: desired group
+    :param key: key of the value that is requested
+    :param value: value to be set for given key
+    :return: Nothing
+    """
+    if group:
+        group = _encode_str(group)
+    c_key = _encode_str(key)
+    if not isinstance(value, float):
+        raise TypeError('"value" parameter must be of type float')
+    c_value = c_double(value)
```

Comment:
Do we need here some approach for a "_ensure_valid_float(value)"

Reply:
No, because floats are saved as an exponentiation so we don't have to worry about the overflow. The typecheck should be enough

Reply:
Uhmm no. The exponentiation is because IEEE 754. In any case should not be required because a Python float is a C double:

```
>>> sys.float_info
sys.float_info(max=1.7976931348623157e+308, max_exp=1024, max_10_exp=308, min=2.2250738585072014e-308, min_exp=-1021, min_10_exp=-307, dig=15, mant_dig=53, epsilon=2.220446049250313e-16, radix=2, rounds=1)
```

The max match the one that from C and C++